src/shady-css/token-cursor.js

```diff
@@ -0,0 +1,36 @@
+const index = Symbol('index');
+const lexer = Symbol('lexer');
+const nextToken = Symbol('nextToken');
+
+class TokenCursor {
```

Comment:
TokenCursor implements peeking. This is extra state that doesn't seem relevant to the Tokenizer.